transport/src/main/java/io/netty/channel/pool/FixedChannelPool.java

```diff
@@ -399,27 +408,31 @@ public final void run() {
 
         @Override
         public void operationComplete(Future<Channel> future) throws Exception {
-            assert executor.inEventLoop();
+            try {
+                assert executor.inEventLoop();
 
-            if (closed) {
-                if (future.isSuccess()) {
-                    // Since the pool is closed, we have no choice but to close the channel
-                    future.getNow().close();
+                if (closed) {
+                    if (future.isSuccess()) {
+                        // Since the pool is closed, we have no choice but to close the channel
+                        future.getNow().close();
+                    }
+                    originalPromise.setFailure(new IllegalStateException("FixedChannelPool was closed"));
+                    return;
                 }
-                originalPromise.setFailure(new IllegalStateException("FixedChannelPool was closed"));
-                return;
-            }
 
-            if (future.isSuccess()) {
-                originalPromise.setSuccess(future.getNow());
-            } else {
-                if (acquired) {
-                    decrementAndRunTaskQueue();
+                if (future.isSuccess()) {
+                    originalPromise.setSuccess(future.getNow());
                 } else {
-                    runTaskQueue();
-                }
+                    if (acquired) {
+                        decrementAndRunTaskQueue();
+                    } else {
+                        runTaskQueue();
+                    }
 
-                originalPromise.setFailure(future.cause());
+                    originalPromise.setFailure(future.cause());
+                }
+            } catch (Throwable cause) {
+                originalPromise.setFailure(cause);
```

Comment:
This should use `tryFailure()` as we may be here if `setFailure()` threw above.

Reply:
Done